src/jvm.rs

```diff
@@ -125,6 +133,104 @@ fn get_or_default_init_jvm() -> crate::GlobalResult<JvmPtr> {
     }
 }
 
+/// Invoked as the body from a JNI native function when it is called by the JVM.
+/// Initializes the environment and invokes `op`. Converts the result into a java
+/// object and returns it. Caller should then return this to the JVM.
+///
+/// # Safety condition
+///
+/// Must be invoked as the entire body of a JNI native function, with
+/// `env` being the `EnvPtr` argument provided.
+pub unsafe fn native_function_returning_object<J, R>(
+    env: EnvPtr<'_>,
+    op: impl FnOnce() -> R,
+) -> jni_sys::jobject
+where
+    J: Upcast<crate::java::lang::Object> + Upcast<J>,
+    R: ToJavaImpl<J> + std::fmt::Debug,
+{
+    init_jvm_from_native_function(env);
+    let old_state = thread::attach_from_jni_callback(env);
+
+    let result = match std::panic::catch_unwind(AssertUnwindSafe(|| op())) {
+        Ok(result) => {
+            let mut jvm = Jvm(env);
+            eprintln!("got result = {result:?}");
+            let obj = result.to_java().execute_with(&mut jvm);
+            eprintln!("obj = {obj:?}");
+            match obj {
+                Ok(Some(p)) => p.as_raw().as_ptr(),
```

Comment:
Here, isn't `obj: Local` deallocated on `Drop` when this Rust function terminates, while it should instead remain allocated to be used from Java? I guess that if the returned object points to a deallocated instance then the JVM might conservatively interpret it as being `null`.

Reply:
*Ah*, yeah, that's prob it, interesting!

Reply:
Good catch @fpoli, I'll tweak that